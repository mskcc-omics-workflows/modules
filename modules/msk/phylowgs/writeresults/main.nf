process PHYLOWGS_WRITERESULTS {
    tag "$meta.id"
    label 'process_medium'

    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'docker://mskcc/phylowgs:v1.5-msk':
        'docker.io/mskcc/phylowgs:v1.5-msk' }"

    input:
    tuple val(meta), path(trees)

    output:
    tuple val(meta), path("*.summ.json.gz")     , emit: summ
    tuple val(meta), path("*.muts.json.gz")     , emit: muts
    tuple val(meta), path("*.mutass.zip")       , emit: mutass
    path "versions.yml"                         , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"

    """
    python2 \\
        /usr/bin/write_results.py \\
        --include-ssm-names \\
        ${prefix} \\
        ${trees} \\
        ${prefix}.summ.json.gz \\
        ${prefix}.muts.json.gz \\
        ${prefix}.mutass.zip
    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        phylowgs: \$PHYLOWGS_TAG
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}.summ.json.gz
    touch ${prefix}.muts.json.gz
    touch ${prefix}.mutass.zip

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        phylowgs: \$PHYLOWGS_TAG
    END_VERSIONS
    """
}
