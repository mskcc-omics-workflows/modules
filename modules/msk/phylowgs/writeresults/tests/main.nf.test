nextflow_process {

    name "Test Process PHYLOWGS_WRITERESULTS"
    script "../main.nf"
    process "PHYLOWGS_WRITERESULTS"

    tag "modules"
    tag "modules_nfcore"
    tag "modules_msk"
    tag "phylowgs"
    tag "phylowgs/writeresults"

    test("phylowgs_writeresults - zip") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map,
                    file(params.test_data_mskcc['neoantigen']['trees'], checkIfExists: true)
                    ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.versions,
                                file(process.out.summ[0][1]).name,
                                file(process.out.muts[0][1]).name,
                                file(process.out.mutass[0][1]).name).match()
                }

            )
        }

    }

    test("phylowgs_writeresults - zip - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map,
                    file('tree_data')
                    ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.versions,
                                file(process.out.summ[0][1]).name,
                                file(process.out.muts[0][1]).name,
                                file(process.out.mutass[0][1]).name).match()
                }

            )
        }

    }


}
