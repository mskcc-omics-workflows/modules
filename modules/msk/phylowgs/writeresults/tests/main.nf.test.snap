{
    "phylowgs_writeresults - zip - stub": {
        "content": [
            [
                "versions.yml:md5,2f91a4391e3a459ac03c94123651b6ed"
            ],
            "test.summ.json.gz",
            "test.muts.json.gz",
            "test.mutass.zip"
        ],
        "timestamp": "2024-06-11T18:02:20.644108"
    },
    "phylowgs_writeresults - zip": {
        "content": [
            [
                "versions.yml:md5,2f91a4391e3a459ac03c94123651b6ed"
            ],
            "test.summ.json.gz",
            "test.muts.json.gz",
            "test.mutass.zip"
        ],
        "timestamp": "2024-06-11T18:02:13.457864"
    }
}